src/main/java/org/wildfly/security/auth/server/IdentityLocator.java

```diff
@@ -0,0 +1,180 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.security.auth.server;
+
+import static org.wildfly.security._private.ElytronMessages.log;
+
+import java.security.Principal;
+
+import org.wildfly.common.Assert;
+import org.wildfly.security.evidence.Evidence;
+
+/**
+ * A locator for an identity.
+ *
+ * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
+ */
+public final class IdentityLocator {
+    private final String name;
+    private final Principal principal;
+    private final Evidence evidence;
+
+    IdentityLocator(final String name, final Principal principal, final Evidence evidence) {
+        this.name = name;
+        this.principal = principal;
+        this.evidence = evidence;
+    }
+
+    /**
+     * Get the identity name.
+     *
+     * @return the identity name (not {@code null})
+     * @throws IllegalStateException if the name field is not set
+     */
+    public String getName() {
+        final String name = this.name;
+        if (name == null) throw log.locatorFieldNotSet("name");
+        return name;
+    }
+
+    /**
+     * Determine whether a name is set.
+     *
+     * @return {@code true} if a name is set, {@code false} otherwise
+     */
+    public boolean hasName() {
+        return name != null;
+    }
+
+    /**
+     * Get the identity principal.
+     *
+     * @return the identity principal (not {@code null})
+     * @throws IllegalStateException if the principal field is not set
+     */
+    public Principal getPrincipal() {
+        final Principal principal = this.principal;
+        if (principal == null) throw log.locatorFieldNotSet("principal");
+        return principal;
+    }
+
+    /**
+     * Determine whether a principal is set.
+     *
+     * @return {@code true} if a principal is set, {@code false} otherwise
+     */
+    public boolean hasPrincipal() {
+        return principal != null;
+    }
+
+    /**
+     * Get the identity evidence.
+     *
+     * @return the identity evidence (not {@code null})
+     * @throws IllegalStateException if the evidence field is not set
+     */
+    public Evidence getEvidence() {
+        final Evidence evidence = this.evidence;
+        if (evidence == null) throw log.locatorFieldNotSet("evidence");
+        return evidence;
+    }
+
+    /**
+     * Determine whether evidence is set.
+     *
+     * @return {@code true} if evidence is set, {@code false} otherwise
+     */
+    public boolean hasEvidence() {
+        return evidence != null;
+    }
+
+    /**
+     * Shortcut method to construct an identity locator from just a name.
+     *
+     * @param name the name (must not be {@code null})
+     * @return the identity locator (not {@code null})
+     */
+    public static IdentityLocator fromName(String name) {
```

Comment:
Could do.  I only added this one because there were really a large number of cases that used name.